### PR TITLE
fix: Improve type prefix and suffix extraction

### LINKED ISSUE #12: TypeID.from_string() validates incorrectly
I am trying to use TypeIDs as API keys and want to include both the key name and the owner name in the prefix, e.g. `keyname_owner_01hxsjs5zqfv39nsdbw9m9qdcp`. While it is possible to create this TypeID, trying to parse the resulting key back into a TypeID fails with an `InvalidTypeIDStringException`. Will submit a PR to fix this shortly.

EDIT: I'm stupid, forgot the quotes. No exploit possible

### PR DESCRIPTION
**Pull request by iron3oxide**
Refactored the logic in `get_prefix_and_suffix` function to improve type prefix and suffix extraction. Now using `rsplit` method to accurately extract the prefix and suffix from the input string. 

Closes #12

### CODE REVIEW

**Comment by akhundMurad** [comment]
Hello @iron3oxide 
Your changes broke the TypeID spec validation.
Please try to resolve them and add additional tests to your case.
Thanks for your contribution!

File: typeid/typeid.py [hunk 1]
```diff
@@ -80,15 +80,14 @@ def from_uuid(suffix: uuid6.UUID, prefix: Optional[str] = None) -> TypeID:
...
+        prefix = None
         suffix = parts[0]
-    elif len(parts) == 2 and parts[0] != "":
-        suffix = parts[1]
-        prefix = parts[0]
-    else:
+    elif len(parts) == 2 and parts[0] == "":
         raise InvalidTypeIDStringException(f"Invalid TypeID: {string}")
+    else:
+        prefix, suffix = parts
```
Comment (khasizadaj) [inline typeid/typeid.py:90]: This doesn't work as it should. As you can see from the exception raised in the test,  it is not `InvalidTypeIDStringException` but `SuffixValidationException`.

It fails when you provide a string with multiple underscores. In that case `part[0]` will never be equal to an empty string and the prefix will be incorrect, in other words, we won't be able to throw an error. 

Better to use `split` and add another condition for length being 3 or more. 

```suggestion
    elif len(parts) >= 3:
        raise InvalidTypeIDStringException(f"Invalid TypeID: {string}")
    else:
        prefix, suffix = parts
```

File: typeid/typeid.py [hunk 2]
```diff
@@ -80,15 +80,14 @@ def from_uuid(suffix: uuid6.UUID, prefix: Optional[str] = None) -> TypeID:
 
 
 def get_prefix_and_suffix(string: str) -> tuple:
-    parts = string.split("_")
-    prefix = None
+    parts = string.rsplit("_", 1)
```
Comment (khasizadaj) [inline typeid/typeid.py:83]: I am suggesting this because of the next comment.

```suggestion
    parts = string.split("_")
```

**Review by khasizadaj — CHANGES_REQUESTED** [review]

**Comment by iron3oxide** [comment]
@khasizadaj I am replying here because it is easier to follow the conversation this way. I don't quite understand your suggested changes.

To preface this: In my opinion, `get_prefix_and_suffix()` should aim to do just that: extract the prefix and suffix candidates from the given string. A special case where it is preferable to raise an error early is a single leading underscore in the given string. 

After that it is up to the `TypeID` class to correctly flag the given prefix and suffix as invalid; which it does, as the tested string in the test you mentioned contains a valid prefix according to the TypeID spec, but no valid suffix. A string such as "___test_invalid_00041061050r3gg28a1c60t3gf" is also correctly flagged as containing an invalid prefix. Could you give me an example string where the changes suggested in this PR produce an unexpected or wrong outcome?

**Comment by khasizadaj** [comment]
Thanks for the clarification. I was thinking that it could be useful to throw exception before that as well, but now I understand your POV as well. We can disregard my comment.

**Comment by akhundMurad** [comment]
@iron3oxide @khasizadaj 
may I resolve the conversation above?

**Comment by iron3oxide** [comment]
Yes, I think it's ready to be merged

**Comment by akhundMurad** [comment]
@iron3oxide @khasizadaj 
thanks, guys!
I will include it in the next release.

**Comment by khasizadaj** [comment]
Yes, it can be closed! Thanks a lot, @iron3oxide.

**Review by akhundMurad — APPROVED** [review]

**Pull request merged**